sdk/include/platform/sunburst/platform-ethernet.hh

```diff
@@ -0,0 +1,639 @@
+
+	/*
+	 * GPIO output pins to be used
+	 */
+	static constexpr uint32_t EthCsPin  = 13;
+	static constexpr uint32_t EthRstPin = 14;
+
+	/**
+	 * The location of registers
+	 */
+	enum class RegisterOffset : uint8_t
+	{
+		/* Chip configuration register */
+		CCR = 0x08,
+		/* MAC address low */
+		MARL = 0x10,
+		/* MAC address middle */
+		MARM = 0x12,
+		/* MAC address high */
+		MARH = 0x14,
+		/* On-chip bus control register */
+		OBCR = 0x20,
+		/* EEPROM control register */
+		EEPCR = 0x22,
+		/* Memory BIST info register */
+		MBIR = 0x24,
+		/* Global reset register */
+		GRR = 0x26,
+
+		/* Wakeup frame registers omitted */
+
+		/* Transmit control register */
+		TXCR = 0x70,
+		/* Transmit status register */
+		TXSR = 0x72,
+		/* Receive control register 1 */
+		RXCR1 = 0x74,
+		/* Receive control register 2 */
+		RXCR2 = 0x76,
+		/* TXQ memory information register */
+		TXMIR = 0x78,
+		/* Receive frame header status register */
+		RXFHSR = 0x7C,
+		/* Receive frame header byte count register */
+		RXFHBCR = 0x7E,
+		/* TXQ command register */
+		TXQCR = 0x80,
+		/* RXQ control register */
+		RXQCR = 0x82,
+		/* TX frame data pointer register */
+		TXFDPR = 0x84,
+		/* RX frame data pointer register */
+		RXFDPR = 0x86,
+		/* RX duration timer threshold register */
+		RXDTTR = 0x8C,
+		/* RX data byte count threshold register */
+		RXDBCTR = 0x8E,
+		/* Interrupt enable register */
+		IER = 0x90,
+		/* Interrupt status register */
+		ISR = 0x92,
+		/* RX frame count and threshold register */
+		RXFCTR = 0x9c,
+		/* TX next total frames size register */
+		TXNTFSR = 0x9E,
+
+		/* MAC address hash table registers omitted */
+
+		/* Flow control low watermark register */
+		FCLWR = 0xB0,
+		/* Flow control high watermark register */
+		FCHWR = 0xB2,
+		/* Flow control overrun watermark register */
+		FCOHR = 0xB4,
+		/* Chip ID and enable register */
+		CIDER = 0xC0,
+		/* Chip global control register */
+		CGCR = 0xC6,
+		/* Indirect access control register */
+		IACR = 0xC8,
+		/* Indirect access data low register */
+		IADLR = 0xD0,
+		/* Indirect access data high register */
+		IADHR = 0xD2,
+		/* Power management event control register */
+		PMECR = 0xD4,
+		/* Go-sleep & wake-up time register */
+		GSWUTR = 0xD4,
+		/* PHY reset register */
+		PHYRR = 0xD4,
+		/* PHY 1 MII-register basic control register */
+		P1MBCR = 0xE4,
+		/* PHY 1 MII-register basic status register */
+		P1MBSR = 0xE6,
+		/* PHY 1 PHY ID low register */
+		PHY1LR = 0xE8,
+		/* PHY 1 PHY ID high register */
+		PHY1HR = 0xEA,
+		/* PHY 1 auto-negotiation advertisement register */
+		P1ANAR = 0xEC,
+		/* PHY 1 auto-negoatiation link partner ability register */
+		P1ANLPR = 0xEE,
+		/* Port 1 PHY special control/status, LinkMD */
+		P1SCLMD = 0xF4,
+		/* Port 1 control register */
+		P1CR = 0xF6,
+		/* Port 1 status register */
+		P1SR = 0xF8,
+	};
+
+	using MACAddress = std::array<uint8_t, 6>;
+
+	enum RxFrameHeaderStatus : uint16_t
```

Comment:
```suggestion
	enum [[clang::flag_enum]] RxFrameHeaderStatus : uint16_t
```

Reply:
I implemented the change but clang-format formats it horribly.